### PR TITLE
Reduce logging in RetryableS3OutputStream

### PR DESCRIPTION
**Pull request by Akshat-Jain**
### Description

This PR reduces logging in RetryableS3OutputStream.

<hr>

This PR has:

- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by cryptoe — APPROVED** [review]

File: extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/RetryableS3OutputStream.java [hunk 1]
```diff
@@ -212,11 +212,10 @@ public void close() throws IOException
     // Closeables are closed in LIFO order
     closer.register(() -> {
       org.apache.commons.io.FileUtils.forceDelete(chunkStorePath);
-      LOG.info("Deleted chunkStorePath[%s]", chunkStorePath);
 
       final long totalBytesUploaded = (currentChunk.id - 1) * chunkSize + currentChunk.length();
       final long totalUploadTimeMillis = pushStopwatch.elapsed(TimeUnit.MILLISECONDS);
-      LOG.info(
+      LOG.debug(
```
Comment (LakshSingla) [inline extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/RetryableS3OutputStream.java]: can you call this method in `log.isDebugEnabled` instead? Computing the log message is expensive and we don't wanna do that if the logging level isn't `DEBUG`

Comment (Akshat-Jain) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/RetryableS3OutputStream.java]: Done

Comment (kfaraz) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/RetryableS3OutputStream.java]: Nit: The log message is formatted only if the logging is needed.

https://github.com/apache/druid/blob/84192b11d72ed195c42827a2e142911f3f87e6ea/processing/src/main/java/org/apache/druid/java/util/common/logger/Logger.java#L128-L133

Checking `isDebugEnabled()` is useful in the Druid code only when one of the args passed to the log itself requires some computational effort. Example:

```
if (log.isDebugEnabled()) {
  log.debug("Log the result of a computation[%s].", someComputation());
}
```

Comment (LakshSingla) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/RetryableS3OutputStream.java]: I see, thanks for mentioning this.

Comment (Akshat-Jain) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/RetryableS3OutputStream.java:218]: @kfaraz Thanks for pointing this out, have made the change!

**Pull request merged**